src/services/HttpService.ts

```diff
@@ -0,0 +1,134 @@
+import 'reflect-metadata';
+import { injectable } from "inversify";
+import i18n from "../locales/config";
+
+export interface HttpService {
+    send<T>(url: string, methodType: MethodType, headers?: ApiHeader, data?: any) : Promise<ApiResponse<T>>;
+}
+
+const baseUrl = process.env.BASE_API_URL
+
+export enum ContentType {
+    FormData,
+    Json
+}
+
+export enum MethodType {
+    POST,
+    GET,
+    PUT,
+    DELETE
+}
+
+export interface ApiResponse<T> {
+    status: number;
+    statusText: string;
+    headers: Headers;
+    data: T;
+}
+
+export interface ApiHeader {
+    contentType?: ContentType;
+    authorization?: string | null;
+}
+
+@injectable()
+export default class DefaultHttpService implements HttpService {
+    private readonly headerKeyContentType = 'Content-Type';
+    private readonly headerValueContentTypeJson = 'application/json';
+    private readonly headerValueContentTypeFormData = 'application/x-www-form-urlencoded';
+    private readonly headerValueMethodTypePost = 'POST';
+    private readonly headerValueMethodTypeGet = 'GET';
+    private readonly headerValueMethodTypePut = 'PUT';
+    private readonly headerValueMehtodTypeDelete = 'DELETE';
+    private readonly headerValueCredentialsTypeInclude = 'include';
+    private readonly headerValueCredentialsTypeOmit= 'omit';
+
+    public async send<T>(url: string, methodType: MethodType, headers?: ApiHeader, data?: any) : Promise<ApiResponse<T>> {
+        const headersRequest = this.getHeaders(headers);
+        const bodyRequest = this.getBody(data, headers);
+        const method = this.getMethod(methodType);
```

Comment:
Isn't this overcodding? We have enum and string const for method types? and one more method with switch/case for define string value to enum??